README.md

```diff
@@ -1,87 +1,121 @@
-The biggest offender though, is that this code violates a principal of beauty, proportion. Let's say we needed to add a new field to our form, like an attribute author_name. Conceptually, that's 1 change. How many changes do we need to make to our form and controller? We need to update the form and then we need to update the controller. One conceptual change should mean one code change. That would be a beatiful proportion.
+* The form has no knowledge of the form, it doesn't know if the form is meant to create or update a record
 
-form_for to the rescue
+* You're forced to have duplicate code throughout the form, it's hard to adhere to DRY principles when utilizing the `form_tag`
 
-form_tag family of helpers - simply generate html, no knowledge, not smart. helpful and shorter than html, but we can do better.
 
-form_for family of helpers are bound to the activerecord object the form relates too. that means every time we build a form tag or an input tag, we can make assumptions based on the model itself. it's really cool, watch.
+## Difference between `form_for` and `form_tag`
 
-form_for(@post) do |form_builder|
+The differences between `form_for` and `form_tag` are subtle, but important, below is a basic breakdown of the differences, we'll start with talking about them at a high level perspective and then get into each one of the aspects on a practical/implementation basis:
 
-end
+* `form_for` binds itself to the model and automatically auto fills the values for each attribute by simply calling the symbol
 
-when we build a form, we pass in the activerecord instance the form is wrapping. form_for will yield an abstract object called a form builder which is an instance of ActionView::FormBuilder to the block we pass it.
+* `form_for` yields an object that you can use to generate your form elements
 
-within the block of form_for, we can use the form_builder to generate inputs and form controls based on the @post object we passed form_for.
+* `form_for` automatically creates the route for the form data as opposed to having to manually declare it
 
-form_for(@post) do |form_builder| <%= form_builder.text_field :title %> end
+* `form_for` gives the option to dynamically change the `submit` button text (this comes in very handy when you're using a form partial and the `new` and `edit` pages will share the same form, but more on that in a later lesson)
 
-that magic line generates the following html for our edit form.
+A good rule of thumb for when to use one approach over the other is below:
 
-we never had to mention @post or post_title or @post.title. everything was automagically generated by calling a method, text_field, on the form_builder object.
+* Use `form_for` when your form is directly connected to a model, extending our example from the introduction, this would be our Hamster's profile edit form that connects to the profile database table
```

Comment:
mention that this is the most common case. (which is especially true for beginners)